### ISSUE TITLE
MS Search - External Items - property added/listed twice

### ISSUE THREAD

**Issue by kasperbolarsen** [microsoft-search/pnp-modern-search#4020]
I have just added the property "customerName" from my external datasource, and it seems like it is added twice, once as customerName and once as resource.properties.customerName

I would expect it to be listed once, and preferably just as customerName

![image](https://github.com/user-attachments/assets/6f02b216-a410-4233-8271-2a904d90e3b8)

**Comment by kasperbolarsen** [comment]
Version 4.13.1
Browser Edge

Steps
1) add Results web part to new page
2) select Microsoft Search as source
3) Set Entity type to "External Items"
4) set Content sources to "AzureSqlConnector3"  
5) check the values in "Selected fields", none of the fields from the Content source is there
6) Check the schema for the Content source in the Search and Intelligence center
7) copy the Source property name for one of the field, in this case CustomerName
8) add CustomerName to Selected Fields 
9) change the layout to Details List
10) select Manage columns
11) open the dropdown for Column value

![image](https://github.com/user-attachments/assets/eec67e2e-6b80-463e-b1ca-8fb963ee3c0f)

**Comment by wobba** [comment]
I assume only the resource version should be kept?

**Comment by gretchunkim** [comment]
When using the external items, I also see what you are describing above. However, I think it's because I have identifed the property information to be pulled (because it is not automatically done so and it's a requirement for external item).  

Have you look into debug mode/view to see if it is listed only once there, correct? This is the case for me. I notice only one of the value works, typically the one that I manually typed/identified in External Items property field.

**Comment by wobba** [comment]
I think it's due to the request name being `Foo` when you add it, and the returned value being `resource.Foo`, and the list puts the two together. Should the web parts hide away the `resource` version?

Not sure how best/easiest to merge the input list you can edit vs the output list used to pick a field as the prop names are different due to how Graph return data.

**Comment by kasperbolarsen** [comment]
no longer relevant as the schema is now extracted and used to populate the Selected Fields dropdown
